src/lib/pipeline/RegisterAliasTable.cc

```diff
@@ -42,6 +42,22 @@ RegisterAliasTable::RegisterAliasTable(
 };
 
 Register RegisterAliasTable::getMapping(Register architectural) const {
+  // Asserts to ensure mapping isn't attempted for an out-of-bound index (i.e.
+  // mapping of WZR / XZR)
+  assert(architectural.type < mappingTable_.size() &&
+         "Invalid register type. Cannot find RAT mapping.");
+  assert(architectural.type >= 0 &&
+         "Invalid register type. Cannot find RAT mapping.");
+  // Validate tags
+  for (int i = 0; i < mappingTable_.size(); i++) {
```

Comment:
Why has this check been added?

Reply:
Comments from lines 45 and 45 `// Asserts to ensure mapping isn't attempted for an out-of-bound index (i.e. mapping of WZR / XZR)`

Reply:
Is this just a sanity check or has this resolved a prior error. I only ask as this logic will have to be triggered for each mapping (may be costly at higher  simulation cycle counts)

Reply:
In essence, it just ensures that only valid tags can be looked up or used as an index - done as a preventative measure after a garbage speculative instruction caused the out of bounds error (tried to map a zero register)

 It may be worth changing these to If statements to catch such an issue in Release also - however the checks added to `Instruction_decode.cc` should prevent the same issue occuring anyway.